src/app/components/NewUserForm/index.tsx

```diff
@@ -0,0 +1,267 @@
+  const [inactive, setInactive] = useState<boolean>(false);
+
+  const formSchema = z.object({
+    username: z
+      .string()
+      .min(2, {
+        message: "O nome de usuário deve ter pelo menos 2 caracteres..",
+      })
+      .nonempty("O nome é obrigatório"),
+    email: z
+      .string()
+      .email("Formato de e-mail inválido")
+      .toLowerCase()
+      .nonempty("O e-mail é obrigatório"),
+    description: z
+      .string()
+      .min(2, {
+        message: "A descrição deve ter pelo menos 2 caracteres..",
+      })
+      .nonempty("O nome é obrigatório"),
+    hardSkills: z.array(z.string()).refine((skills) => skills.length > 0, {
+      message: "Selecione pelo menos uma habilidade",
+    }),
+    squad: z.string().nonempty("Selecione a squad"),
+    biography: z.optional(z.string()),
+    inactiveUser: z.optional(z.boolean()),
+    photo: z.custom<File | null>(),
+  });
+
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      photo: null,
+      inactiveUser: false,
+      biography: "",
+      username: "",
+      email: "",
+      description: "",
+      hardSkills: [],
+      squad: "",
+    },
+  });
+
+  const handleImageChange = (file: File) => {
+    form.setValue("photo", file);
+  };
+
+  const handleRemoveImage = () => {
+    form.setValue("photo", null);
+  };
+  const onChangeHardSkills = (selectedOptions: MultiValueProps) => {
+    const options = selectedOptions.map((option) => option.label);
+    form.setValue("hardSkills", options);
+  };
+
+  const onChangeInactiveUser = () => {
+    setInactive((prevState) => !prevState);
+    form.setValue("inactiveUser", !inactive);
+  };
+
+  function onSubmit(values: z.infer<typeof formSchema>) {
+    console.log(values);
+  }
+
+  return (
+    <div className={Styles.container()}>
+      <Form {...form}>
+        <form
+          onSubmit={form.handleSubmit(onSubmit)}
+          className={Styles.formContainer()}
+        >
+          <div className={Styles.formContent()}>
+            <div className={Styles.contentLeftFields()}>
+              <FormField
+                name="photo"
+                render={({ field }) => (
+                  <FormItem>
+                    <FormLabel className={Styles.photo()}>
+                      Adicione uma foto:
+                    </FormLabel>
+                    <FormControl>
+                      <ImageUpload
+                        {...field}
+                        onChange={(file) => handleImageChange(file)}
+                        onRemove={handleRemoveImage}
+                        accept="image/*"
+                      />
+                    </FormControl>
+                  </FormItem>
+                )}
+              />
+              <FormField
+                name="inactiveUser"
+                render={({ field }) => (
+                  <FormItem>
+                    <FormControl>
+                      <Switch
```

Comment:
Essa parte de ter que setar o value no form, vai sempre se repitir, né?! Será que não conseguimos criar um SwitchControl e forçar isso? Seguiria o padrão dos outros componentes, eles fariam o acesso ao form, através do Context.